20230312/mike.java

```diff
@@ -0,0 +1,45 @@
+/*
+Just like the concept of MergeSort.
+Partition every node and let it merge in increased.
+After merge all node, return it.
+Tomorrow I'll see other way to solve this problem, but I don't have energy to do this know.
+*/
+class Solution {
+    public static ListNode mergeKLists(ListNode[] lists){
+        return partion(lists,0,lists.length-1);
+    }
+
+    public static ListNode partition(ListNode[] lists,int start,int end){
+        if(start==end) {
+            return lists[start];
+        }
+        
+        if(start < end) {
+            int q=(start+end)/2;
+            ListNode l1 = partition(lists,start,q);
+            ListNode l2 = partition(lists,q+1,end);
+            
+            return merge(l1,l2);
```

Comment:
hmm... how about just merging linked lists two by two instead of the complex logic?

let's say, we have 6 linked lists indexed from 0 to 5. the merging sequence is like.
1. 0 + 1
2. 2 + 3
3. 4 + 5
4. 0/1 + 2/3
5. 0/1/2/3 + 4/5
6. finally, we have 0/1/2/3/4/5.

Reply:
Sounds great. I'll try it tonight.